model.py

```diff
@@ -3284,74 +3258,55 @@ def all_cover_images(self):
 
     def all_descriptions(self):
         _db = Session.object_session(self)
-        primary_identifier_ids = [
-            x.primary_identifier.id for x in self.editions]
+        primary_identifier_id = self.presentation_edition.primary_identifier.id
         data = Identifier.recursively_equivalent_identifier_ids(
-            _db, primary_identifier_ids, 5, threshold=0.5)
+            _db, [primary_identifier_id], 5, threshold=0.5)
         flattened_data = Identifier.flatten_identifier_ids(data)
         return Identifier.resources_for_identifier_ids(
             _db, flattened_data, Hyperlink.DESCRIPTION).filter(
                 Resource.content != None).order_by(
                 Resource.quality.desc())
 
 
-    def set_primary_edition(self, new_primary_edition):
-        """ Sets primary edition and lets owned pools and editions know.
+    def set_presentation_edition(self, new_presentation_edition):
+        """ Sets presentation edition and lets owned pools and editions know.
             Raises exception if edition to set to is None.
         """
         # only bother if something changed, or if were explicitly told to 
         # set (useful for setting to None)
-        if not new_primary_edition:
-            error_message = "Trying to set primary_edition to None on Work [%s]" % self.id
+        if not new_presentation_edition:
+            error_message = "Trying to set presentation_edition to None on Work [%s]" % self.id
             raise ValueError(error_message)
 
-        self.primary_edition = new_primary_edition
-
-        # go through the loser pools, and tell them they lost
-        for pool in self.license_pools:
-            if pool.presentation_edition is self.primary_edition:
-                # make sure edition knows it's primary
-                pool.presentation_edition.is_primary_for_work = True
-            else:
-                pool.mark_edition_primarity(primary_for_work_edition=None)
+        self.presentation_edition = new_presentation_edition
 
+        # Let the edition's license pool know it has a work.
+        if self.presentation_edition.is_presentation_for:
+            self.presentation_edition.is_presentation_for.work = self
```

Comment:
Right, loser editions won't have a work already. The old code had to set is_primary_for_work separately.